src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayerMP.java

```diff
@@ -897,4 +902,20 @@ public void setSpectatorTarget(@Nullable org.spongepowered.api.entity.Entity ent
         return event;
     }
 
+    @Override
+    public WorldBorder getWorldBorder() {
+        if (this.worldBorder == null) {
+            this.worldBorder = (WorldBorder) this.world.getWorldBorder();
+        }
+        return this.worldBorder;
+    }
+
+    @Override
+    public void setWorldBorder(WorldBorder border, Cause cause) {
+        WorldBorder originalBorder = this.worldBorder;
+        this.worldBorder = border;
+        this.connection.sendPacket(new SPacketWorldBorder((net.minecraft.world.border.WorldBorder) (Object) border, SPacketWorldBorder.Action.INITIALIZE));
+        SpongeImpl.postEvent(SpongeEventFactory.createChangeWorldBorderEventTargetPlayer(cause, originalBorder, border, this));
```

Comment:
The event is cancellable but you don't handle the case where it's cancelled. same goes for below too.